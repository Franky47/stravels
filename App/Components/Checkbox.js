import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Image,
  TouchableHighlight,
  View,
  ViewPropTypes
} from 'react-native'
import styles from './Styles/CheckboxStyle'

export default class Checkbox extends Component {
  // Prop type warnings
  static propTypes = {
    ...(ViewPropTypes || View.PropTypes),
    checked: PropTypes.bool,
    color: PropTypes.func,
    onPress: PropTypes.func.isRequired,
    onLongPress: PropTypes.func
  }

  // Defaults for props
  static defaultProps = {
    checked: false,
    color: () => 'black',
    onLongPress: () => {}
  }

  render () {
    const checked = this.props.checked
    const imgSource = checked ? require('../Images/checkbox-on.png')
                              : require('../Images/checkbox-off.png')

    return (
      <TouchableHighlight
        onPress={this.props.onPress}
        onLongPress={this.props.onLongPress}
        underlayColor='transparent'
      >
        <View style={styles.container}>
          <Image source={imgSource} style={{tintColor: this.props.color(checked)}} />
        </View>
      </TouchableHighlight>
    )
  }
}
