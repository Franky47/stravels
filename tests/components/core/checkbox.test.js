import 'react-native'
import React from 'react'
import Checkbox from '@stravels/components/core/checkbox'
// import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Check / Uncheck', () => {
  test('unchecked', () => {
    const tree = renderer.create(<Checkbox />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('checked', () => {
    const tree = renderer.create(<Checkbox checked />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Custom Colors', () => {
  test('enabled - unchecked', () => {
    const tree = renderer.create(<Checkbox colorOff='#f00' />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('enabled - checked', () => {
    const tree = renderer.create(<Checkbox colorOn='#f00' checked />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('disabled - unchecked', () => {
    const tree = renderer.create(<Checkbox colorDisabledOff='#f00' />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('disabled - checked', () => {
    const tree = renderer.create(<Checkbox colorDisabledOn='#f00' checked />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Size support', () => {
  test('unchecked', () => {
    const tree = renderer.create(<Checkbox size={42} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('checked', () => {
    const tree = renderer.create(<Checkbox size={42} checked />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

// describe('onPress', () => {
//   test('should not work if disabled', () => {
//     const mock = jest.fn()
//     const checkbox = shallow(<Checkbox onPress={mock} disabled />)
//     expect(checkbox.prop('onPress')).toEqual(mock)
//     expect(mock).not.toHaveBeenCalled()
//     checkbox.simulate('press')
//     expect(mock).not.toHaveBeenCalled()
//   })
//   test('should work if enabled', () => {
//     const mock = jest.fn()
//     const checkbox = shallow(<Checkbox onPress={mock} />)
//     expect(checkbox.prop('onPress')).toEqual(mock)
//     expect(mock).not.toHaveBeenCalled()
//     checkbox.simulate('press')
//     expect(mock).toHaveBeenCalled()
//   })
// })
