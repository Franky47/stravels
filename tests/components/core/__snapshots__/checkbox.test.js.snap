// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check / Uncheck checked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#47cc5e",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Check / Uncheck unchecked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "grey",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Custom Colors disabled - checked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#47cc5e",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Custom Colors disabled - unchecked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "grey",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Custom Colors enabled - checked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#f00",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Custom Colors enabled - unchecked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#f00",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Size support checked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#47cc5e",
        "fontSize": 42,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`Size support unchecked 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "grey",
        "fontSize": 42,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
